### PR TITLE
feat: SIGNL4 provider added

### LINKED ISSUE #734: Provider Request: SIGNL4
**Describe the provider you want to add**
[SIGNL4](https://www.signl4.com/) is a mobile alerting and incident response service for operational teams. It notifies responsible users and teams about critical situations quickly and reliable via app push, SMS text, and voice call including escalation, on-call duty scheduling, collaboration and more.

**Describe your use case**
Our customers connect to various system (https://www.signl4.com/webhook-rest-smtp-api-integration/#integration) from IT, SecOps, but also IoT and production. They sometimes need a consolitation layer for alert.

**Are you already using Keep?**
No

**Additional context**
We provide SIGNL4 and will be happy to provide the Python code for the provider.

### PR DESCRIPTION
**Pull request by rons4**
SIGNL4 provider added for mobile alerting and incident response.

Closes #734

## 📑 Description

SIGNL4 is a mobile alerting and incident response service for operational teams. The SIGNL4 provider allows to alert users or teams about alerts in Keep via app push, SMS text, voice call, including escalation, on-call scheduling, etc.

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [x] My code requires changes to the documentation
- [x] I have updated the documentation as required
- [x] All the tests have passed

## ℹ Additional Information

I have tested the SIGNL4 provider in my environment and all seems to work fine.
If there are any comments, please just let me know.

### CODE REVIEW

File: keep/providers/signl4_provider/signl4_provider.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+import requests
+
+from keep.contextmanager.contextmanager import ContextManager
+from keep.providers.base.base_provider import BaseProvider
+from keep.providers.models.provider_config import ProviderConfig
+from keep.providers.providers_factory import ProvidersFactory
+
+class S4Status(str, enum.Enum):
+    """
+    Ilert incident status.
```
Comment (shahargl) [inline keep/providers/signl4_provider/signl4_provider.py:15]: Ilert?

File: keep/providers/signl4_provider/signl4_provider.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    """
+
+    NEW = "new"
+    ACKNOWLEDGES = "acknowledged"
```
Comment (shahargl) [inline keep/providers/signl4_provider/signl4_provider.py:19]: typo? ACKNOWLEDGES => ACKNOWLEDGED?

File: keep/providers/signl4_provider/signl4_provider.py [hunk 3]
```diff
@@ -0,0 +1,149 @@
+import dataclasses
```
Comment (shahargl) [inline keep/providers/signl4_provider/signl4_provider.py:1]: the recommendation for providers is to add "validate_scopes" function that is triggered when the user clicks "connect" and let them know if the connection/authentication succeeded.

File: keep/providers/signl4_provider/signl4_provider.py [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+
+        # SIGNL4 webhook URL
+        webhook_url = 'https://connect.signl4.com/webhook/' + self.authentication_config.signl4_integration_secret
+
+        try:
+            result = requests.post(url = webhook_url, json = alert_data)
+
+            if result.status_code == 201:
+                # Success
+                print(result.text)
```
Comment (shahargl) [inline keep/providers/signl4_provider/signl4_provider.py:109]: logger instead of printing

**Review by shahargl — COMMENTED** [review]
left some comments overall looks great!

File: docs/providers/documentation/signl4-provider.mdx [hunk 5]
```diff
@@ -0,0 +1,48 @@
+---
```
Comment (shahargl) [inline docs/providers/documentation/signl4-provider.mdx:1]: can you please add this page to `mint.json`?
CleanShot 2024-01-24 at 11 28 26@2x

File: keep/providers/signl4_provider/signl4_provider.py (hunk 4 above)
Comment (rons4) [reply keep/providers/signl4_provider/signl4_provider.py]: Thanks and changed.

File: keep/providers/signl4_provider/signl4_provider.py (hunk 1 above)
Comment (rons4) [reply keep/providers/signl4_provider/signl4_provider.py]: Oops. Changed.

File: keep/providers/signl4_provider/signl4_provider.py (hunk 2 above)
Comment (rons4) [reply keep/providers/signl4_provider/signl4_provider.py:19]: Thanks and corrected.

File: docs/providers/documentation/signl4-provider.mdx (hunk 5 above)
Comment (rons4) [reply docs/providers/documentation/signl4-provider.mdx:1]: Done.

**Pull request merged**